### PR TITLE
SAK-47172 only send submission notification to active members in site

### PR DESCRIPTION
**Pull request by ottenhoff**

### CODE REVIEW

File: assignment/impl/src/java/org/sakaiproject/assignment/impl/AssignmentServiceImpl.java [hunk 1]
```diff
@@ -4656,17 +4655,31 @@ private void notificationToInstructors(AssignmentSubmission submission, Assignme
...
+            Set<String> siteUsers = null;
+            try {
+                siteUsers = siteService.getSite(siteId).getUsers();
+            } catch (IdUnusedException e) {
+                log.warn("Could not find site: {}", siteId, e);
+            }
+            Set<User> users = submission.getSubmitters()
+                    .stream()
+                    .map(AssignmentSubmissionSubmitter::getSubmitter)
+                    .filter(siteUsers::contains) // this filters active users
```
Comment (ottenhoff) [inline assignment/impl/src/java/org/sakaiproject/assignment/impl/AssignmentServiceImpl.java:4670]: this is the only major change in the PR

**Review by bjones86 — APPROVED** [review]

File: assignment/impl/src/java/org/sakaiproject/assignment/impl/AssignmentServiceImpl.java (hunk 1 above)
Comment (bjones86) [reply assignment/impl/src/java/org/sakaiproject/assignment/impl/AssignmentServiceImpl.java:4670]: Paul just pointed this out to me: siteUsers could still be `null` here in the event that an exception is generated in the above try/catch

File: assignment/impl/src/java/org/sakaiproject/assignment/impl/AssignmentServiceImpl.java [hunk 2]
```diff
@@ -4656,17 +4655,31 @@ private void notificationToInstructors(AssignmentSubmission submission, Assignme
...
-            Set<User> users = submitterIds.stream().map(id -> {
-                try {
-                    return userDirectoryService.getUser(id);
-                } catch (UserNotDefinedException e) {
-                    log.warn("Could not find user with id = {}, {}", id, e.getMessage());
-                }
-                return null;
-            }).filter(Objects::nonNull).collect(Collectors.toSet());
+            String siteId = assignment.getContext();
+            Set<String> siteUsers = null;
```
Comment (bjones86) [inline assignment/impl/src/java/org/sakaiproject/assignment/impl/AssignmentServiceImpl.java:4661]:
```suggestion
            Set<String> siteUsers = Collections.emptySet();
```

File: assignment/impl/src/java/org/sakaiproject/assignment/impl/AssignmentServiceImpl.java [hunk 3]
```diff
@@ -4656,17 +4655,31 @@ private void notificationToInstructors(AssignmentSubmission submission, Assignme
...
+            try {
+                siteUsers = siteService.getSite(siteId).getUsers();
+            } catch (IdUnusedException e) {
+                log.warn("Could not find site: {}", siteId, e);
+            }
+            Set<User> users = submission.getSubmitters()
+                    .stream()
+                    .map(AssignmentSubmissionSubmitter::getSubmitter)
+                    .filter(siteUsers::contains) // this filters active users
+                    .collect(Collectors.toSet())
```
Comment (plukasew) [inline assignment/impl/src/java/org/sakaiproject/assignment/impl/AssignmentServiceImpl.java:4671]: Is collecting here necessary?

Comment (ottenhoff) [reply assignment/impl/src/java/org/sakaiproject/assignment/impl/AssignmentServiceImpl.java:4671]: yeah i tried to remove it

File: assignment/impl/src/java/org/sakaiproject/assignment/impl/AssignmentServiceImpl.java (hunk 1 above)
Comment (ottenhoff) [reply assignment/impl/src/java/org/sakaiproject/assignment/impl/AssignmentServiceImpl.java:4670]: thanks

File: assignment/impl/src/java/org/sakaiproject/assignment/impl/AssignmentServiceImpl.java (hunk 3 above)
Comment (plukasew) [reply assignment/impl/src/java/org/sakaiproject/assignment/impl/AssignmentServiceImpl.java:4671]: That's odd, filter() just returns a stream anyway. My IDE doesn't complain if I remove the collect.

Comment (ottenhoff) [reply assignment/impl/src/java/org/sakaiproject/assignment/impl/AssignmentServiceImpl.java:4671]: ahhh remove collect and the stream.... ok done

**Pull request merged**